collections/_posts/2024-04-11-Repaint-diffusion-model-inpainting.md

```diff
@@ -0,0 +1,164 @@
+* Free-form inpainting, i.e., works with a wide variety of masks
+* Inpainting is done during the inference stage, by modifying the generation process, thus not requiring specific training
+* The Repaint method outperforms state-of-the-art autoregressive and GAN approaches for at least five out of the six mask distributions tested
+
+&nbsp;
+
+# Introduction
+
+The state-of-the art of inpainting methods mostly use a distribution of arbitrary masks to train their models. Resulting in poor generalization capabilities. The paper proposes a new method to condition the generation process. The Repaint method takes place during the reverse diffusion and does not require specific training; a classic DDPM is enough!
+
+![](/collections/images/Repaint/masks_used.jpg)
+
+<center style="font-style:italic">
+Figure 1: Illustration of the variety of masks used.
+</center>
+
+&nbsp;
+
+# Method
+## Inpainting algorithm and general idea
+
+A quick reminder about diffusion models is needed to understand the following parts: Diffusion models are basically U-net networks, which, through training, try to predict the noise added to an image. We define a range of noise levels from $$0$$ to $$t$$, $$x_0$$ denoting an input image with no noise added and $$x_t$$ denoting a noisy image $$\sim \mathcal{N}(0, I)$$. The process of gradually adding noise to an image is called the **forward diffusion process**, and the action of gradually removing it is called the **reverse diffusion process**.
+The forward diffusion process is modelized as a Markov chain with noise added at each step, we can use the following equation to generate a picture at a specific noise level:
+
+&nbsp;
+
+$$
+q(x_t|x_0) = \mathcal{N}(x_t; \sqrt{\bar{\alpha_t}}x_0; (1 − \bar{\alpha_t})I)\\
+$$
+
+&nbsp;
+
+As previously mentioned, the innovation of the Repaint paper is due to its method of generation, which is conditioned by the known region of the image. They introduce some notations: the ground truth image is denoted as $$x$$, the unknown pixels as $$m \odot x$$ and the known pixels as $$(1 − m) \odot x$$. They define their new reverse diffusion process as follows:
+
+&nbsp;
+
+$$
+\begin{aligned}
+x^{known}_{t−1} & ∼ \mathcal{N} (\sqrt{\bar{\alpha_t}}x_0, (1 − \bar{\alpha_t})I)\\
+x^{unknown}_{t−1} & ∼ \mathcal{N} (\mu_\theta(x_t, t), \Sigma_\theta(x_t, t))\\
+x_{t−1} & = m \odot x^{known}_{t−1} + (1 − m) \odot x^{unknown}_{t−1}
+\end{aligned}
+$$
+
+&nbsp;
+
+Basically, the idea boils down to sampling from known and unknown regions at a given timestep. Since this sampling is done through all steps of the reverse diffusion process, the distribution of the intermediaries $$x_{t-1}$$ images matches the propriety of the distributions from a regular reverse diffusion process, thus filling the unknown region with a texture that matches the neighboring regions. The final result is an image with the known regions intact, and the unknown region filled with generated texture.
```

Comment:
I tried some rephrasing, like I explained in my other comment, the idea of the method is linked to the bayesian nature of the algorithm.

Reply:
finally rephrased into :
`Basically, the idea boils down to sampling from known and unknown regions at a given timestep. Since this sampling is done through all steps of the reverse diffusion process, the distribution of the intermediary $$x_{t-1}$$ images match more and more the properties of the known image at each denoising steps. At the end of the process, the generated image is an image with **intact** known region, and a texture related to the known region which covers the unknown region.`